### PR TITLE
[SHELL32] Improve Start-Run Dialog Box Features

### PR DESCRIPTION
**Pull request by Doug-Lyons**
## Fix some Start-Run Dialog Box functions that previously failed.

_Change code in dialogs.cpp to try two different methods for ShellExec._

JIRA issue: [CORE-16898](https://jira.reactos.org/browse/CORE-16898)

## Changes to dialogs.cpp to improve ShellExecute ability.

_Fixes "iexplore" and "www.Web-Site-Here.xxx"._

Testbot results:

KVM: https://reactos.org/testman/compare.php?ids=78129,78134, LGTM
 VBox: https://reactos.org/testman/compare.php?ids=78131,78135, LGTM

### CODE REVIEW

**Review by JoachimHenze — APPROVED** [review]
Approved. See my testing table here:
https://jira.reactos.org/browse/CORE-16898?focusedCommentId=128749&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-128749

File: dll/win32/shell32/dialogs/dialogs.cpp [hunk 1]
```diff
@@ -670,9 +686,20 @@ static INT_PTR CALLBACK RunDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARA
 
                         case RF_OK:
                             if (SUCCEEDED(ShellExecCmdLine(hwnd, pszExpanded, pszStartDir, SW_SHOWNORMAL, NULL,
-                                                           SECL_ALLOW_NONEXE)))
+                            /* We use SECL_NO_UI because we don't want to see
+                             * errors here, but we will try again below and
+                             * there we will output our errors. */
```
Comment (HBelusca) [inline dll/win32/shell32/dialogs/dialogs.cpp:691]: Move the comment before the function call.

Comment (Doug-Lyons) [reply dll/win32/shell32/dialogs/dialogs.cpp:690]: Done. Thanks.

**Review by katahiromz — APPROVED** [review]
I think the bug is in `ShellExecCmdLine`.
This PR is a valid workaround.

**Pull request merged**